robottelo/host_helpers/api_factory.py

```diff
@@ -628,12 +628,6 @@ def __init__(self, temp):
         :param entities.ProvisioningTemplate temp: entities.ProvisioningTemplate's object
         """
         self.temp = temp
```

Comment:
if not isinstance(self.temp, entities.ProvisioningTemplate):
            raise TypeError(
                'The template should be of type entities.ProvisioningTemplate, {} given'.format(
                    type(temp)
                )
            )

Reply:
but `entities` is not defined or is it?

Reply:
You can import it.

Reply:
@shweta83 Could you please elaborate on why we need to check the object type? I am inquisitive to know why we need it here considering that we do not check types in other places.

Reply:
As long as the patch works, I am fine.